manager/src/main/java/org/openremote/manager/dashboard/DashboardStorageService.java

```diff
@@ -0,0 +1,182 @@
+        container.getService(ManagerWebService.class).addApiSingleton(
+                new DashboardResourceImpl(
+                        container.getService(TimerService.class),
+                        identityService,
+                        this,
+                        container.getService(MessageBrokerService.class)
+                )
+        );
+    }
+
+    @Override
+    public void start(Container container) throws Exception {
+        /* code not overridden yet */
+    }
+
+    @Override
+    public void stop(Container container) throws Exception {
+        /* code not overridden yet */
+
+    }
+
+
+    /* --------------------------  */
+
+    // Getting ALL dashboards from a realm
+    protected <T extends Dashboard> Dashboard[] findAllOfRealm(String realm, String userId) {
+        return this.findAllOfRealm(realm, userId, false);
+    }
+
+    @SuppressWarnings("java:S2326")
+    protected <T extends Dashboard> Dashboard[] findAllOfRealm(String realm, String userId, Boolean editable) {
+        Object[] result = persistenceService.doReturningTransaction(em -> {
+            try {
+                CriteriaBuilder cb = em.getCriteriaBuilder();
+                CriteriaQuery<Dashboard> cq = cb.createQuery(Dashboard.class);
+                Root<Dashboard> root = cq.from(Dashboard.class);
+                List<Predicate> predicates = new ArrayList<>();
+                if(realm != null) {
+                    predicates.add(cb.like(root.get("realm"), realm));
+                }
+                if(editable) {
+                    predicates.add(cb.or(
+                            root.get("editAccess").in(DashboardAccess.PUBLIC, DashboardAccess.SHARED),
+                            cb.and(root.get("editAccess").in(DashboardAccess.PRIVATE), root.get("ownerId").in(userId))
+                    ));
+                } else {
+                    predicates.add(cb.or(
+                            root.get("viewAccess").in(DashboardAccess.PUBLIC, DashboardAccess.SHARED),
+                            cb.and(root.get("viewAccess").in(DashboardAccess.PRIVATE), root.get("ownerId").in(userId))
+                    ));
+                }
+                CriteriaQuery<Dashboard> all = cq.select(root).where(predicates.toArray(new Predicate[]{}));
+                        //.where(cb.and(root.get("viewAccess").in(DashboardAccess.PRIVATE), root.get("ownerId").in(userId)));
+                TypedQuery<Dashboard> allQuery = em.createQuery(all);
+                Dashboard[] dashboards = allQuery.getResultList().toArray(new Dashboard[0]);
+                return dashboards;
+
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+            return new ArrayList<Dashboard>().toArray(); // Empty array if nothing found.
+        });
+        // Object[] to Dashboard[]
+        return Arrays.copyOf(result, result.length, Dashboard[].class);
```

Comment:
This shouldn't be necessary and doReturningTransaction should return the correct type